issue.go

```diff
@@ -96,29 +96,68 @@ type Epic struct {
 	Done    bool   `json:"done" structs:"done"`
 }
 
-// A description field may be a plain old string or a document structure. This
-// is extremely annoying as they have the same field name and cannot both unmarshal
-// to the same type without jumping through hoops.
-type Document string
+type Document struct {
+	Version int        `json:"version,omitempty"`
+	Type    string     `json:"type,omitempty"`
+	Content ContentSet `json:"content,omitempty"`
+}
+
+type Content struct {
+	Type    string     `json:"type,omitempty"`
+	Text    string     `json:"text,omitempty"`
+	Content ContentSet `json:"content,omitempty"`
+}
+
+const (
+	Doc  = "doc"
+	Text = "text"
+)
+
+type ContentSet []Content
+
+func (c ContentSet) String() string {
+	b := &strings.Builder{}
+	for i, e := range c {
+		if i > 0 {
+			b.WriteString("  ")
+		}
+		b.WriteString(e.String())
+	}
+	return b.String()
+}
+
+func (c Content) String() string {
+	return c.Text + " " + c.Content.String()
```

Comment:
I guess this should only include the separator space if both other sides have content, we probably should check for that. Maybe something like:

```go
b := &strings.Builder{}
if c.Text != "" {
  b.WriteString(c.Text)
}
if v := c.Content.String(); v != "" {
  if b.Len() > 0 {
    b.WriteString(" ")
  }
  b.WriteString(v)
}
return b.String()
```